tests/kernel/wave/common/utils.py

```diff
@@ -75,14 +76,33 @@ def scaled_dot_product_attention_bhsd(
     scale: float = query.shape[-1] ** -0.5
     attn_logits: Tensor = torch.matmul(query, key.transpose(-2, -1)) * scale
 
+    if sliding_window >= 0:
+        assert is_causal, f"Sliding window only supported with causal"
+
     if is_causal:
         seq_len_q, seq_len_k = attn_logits.shape[-2], attn_logits.shape[-1]
-        causal_mask: Tensor = torch.tril(
-            torch.ones(
-                (seq_len_q, seq_len_k), device=attn_logits.device, dtype=torch.bool
+        if sliding_window < 0:
```

Comment:
any thoughts on dropping the if-else condition in favor of `if (sliding_window >= 0) existing_mask.triu(-window_size)`?

Reply:
yeah good point, that will simplify the code.